qemu/tests/disk_extension.py

```diff
@@ -0,0 +1,168 @@
+    increment_size = int(params["increment_size"][0:-1])
+    size_unit = params["increment_size"][-1]
+    guest_cmd = params["guest_cmd"]
+
+    loop_device_backend_img_tag = params["loop_device_backend_img_tag"]
+    loop_device_img_tag = params["loop_device_img_tag"]
+
+    loop_device = params["loop_device"]
+
+    loop_device_backend_img_param = params.object_params(
+        loop_device_backend_img_tag)
+    loop_device_img_param = params.object_params(loop_device_img_tag)
+    drive_format = loop_device_img_param["drive_format"]
+    image_aio = loop_device_img_param["image_aio"]
+    tmpfs_folder = params.get("tmpfs_folder", "/tmp/xtmpfs")
+
+    if loop_device_backend_img_param["image_format"] != "raw":
+        test.cancel("Wrong loop device backend image format in config file.")
+
+    logging.debug("current_size:{}, max_size:{}"
+                  .format(current_size, max_size))
+    logging.debug("increment_size:{}".format(increment_size))
+    logging.debug("data disk drive_format:{} image_aio:{}".format(drive_format,
+                                                                  image_aio))
+    error_context.context("Start to setup tmpfs folder", logging.info)
+    prepare_tmpfs_folder(tmpfs_folder)
+
+    error_context.context("Start to create image on loop device", logging.info)
+    loop_device_backend_img = QemuImg(loop_device_backend_img_param,
+                                      data_dir.get_data_dir(),
+                                      loop_device_backend_img_tag)
+    loop_device_img = QemuImg(loop_device_img_param, data_dir.get_data_dir(),
+                              loop_device_img_tag)
+    create_image_on_loop_device(loop_device_backend_img, loop_device_img)
+
+    try:
+        # start to boot vm
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        error_context.context("Get the main VM", logging.info)
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+
+        timeout = int(params.get("login_timeout", 360))
+        session = vm.wait_for_login(timeout=timeout)
+
+        vm.verify_status("running")
+
+        disk_serial = params.get("disk_serial", "not_define")
+        disk_path = get_linux_drive_path(session, disk_serial)
+        if not disk_path:
+            test.fail("Can not find %s in guest." % disk_serial)
+
+        logging.debug("Find disk path %s for %s" % (disk_path, disk_serial))
+        guest_cmd = guest_cmd % disk_path
+        logging.debug("guest_cmd: %s" % guest_cmd)
+
+        wait_timeout = int(params.get("wait_timeout", 3))
+        logging.debug("Wait timeout is '%d' seconds", wait_timeout)
+
+        guest_cmd_output = session.sendline(guest_cmd)
+        logging.info(guest_cmd_output)
+
+        try:
+            while current_size < max_size:
+                current_size += increment_size
+                current_size_string = str(current_size) + size_unit
+
```

Comment:
I think the line159~162 codes should be move here(line 149) since firstly to check the status of VM then resize the image from docsting:
```
    6) Access  guest vm and execute dd operation on the data disk.
     the IO size is same as the loop device virtual disk size.
    7) Verify vm status is io-error status in hmp.
    8) Continue to increase disk size of the raw image file,
     and update the loop device.
```